editors/emacs/Portfile

```diff
@@ -107,7 +107,7 @@ if {$subport eq "emacs-devel" || $subport eq "emacs-app-devel"} {
     revision        1
 
     fetch.type      git
-    git.url         --depth=1000 https://github.com/emacs-mirror/emacs.git
+    git.url         --shallow-since=2020-11-26 https://github.com/emacs-mirror/emacs.git
```

Comment:
It is quite new feature and old macOS may not support this that will broke this port.

Details: https://github.com/git/git/commit/e34de73c5653c713e6edddc5a12c8befc50d311f

Reply:
I don't understand the relevance of your link; `--shallow-since` was added in git 2.11.0, as I mentioned earlier. That was released in November 2016.

I saw that `port rdeps emacs-devel` included the git port, which is guaranteed to be new enough to support `--shallow-since`. However looking under the hood I see that `rdeps` may be misleading:

https://github.com/macports/macports-base/blob/f613d8a9d1be0d1fd0c1466b2eb8622dcdfd0167/src/port1.0/portfetch.tcl#L197-L203

So there is a window of OS versions from Mavericks (10.9; released June 2013) to whenever the bundled git became 2.11.0 (no idea; my Catalina installation has 2.24.3) that would fail.

We could make it work by always depending on the git port, but that seems overkill. I'll revert this part.

Reply:
@amake honestly I love idea of `--shallow-since` to be honest. Anyway, I've testes `--depth 100` and `--depth 1000` and differences was a few MiB that isn't something measured to care about it.

But I see disadvantage of `--shallow-since`: right now upgrade of dev port needs to be done at one place => update git commit.

If `--deep` to small, it just fail on compile and someone defently will report it.

When a way with `--shallow-since` introduced a one more magic constant what need to be updated otherwise each day the repo will consume more and more traffic.